src/libsystemd/sd-journal/journal-verify.c

```diff
@@ -1135,19 +1139,37 @@ int journal_file_verify(
 
 #if HAVE_GCRYPT
                         if (JOURNAL_HEADER_SEALED(f->header)) {
-                                uint64_t q, rt;
+                                uint64_t q, rt, rt_end;
 
                                 debug(p, "Checking tag %"PRIu64"...", le64toh(o->tag.seqnum));
 
                                 rt = f->fss_start_usec + le64toh(o->tag.epoch) * f->fss_interval_usec;
-                                if (entry_realtime_set && entry_realtime >= rt + f->fss_interval_usec) {
+                                rt_end = usec_add(rt, f->fss_interval_usec);
+                                if (entry_realtime_set && entry_realtime >= rt_end) {
                                         error(p,
                                               "tag/entry realtime timestamp out of synchronization (%"PRIu64" >= %"PRIu64")",
                                               entry_realtime,
                                               rt + f->fss_interval_usec);
                                         r = -EBADMSG;
                                         goto fail;
                                 }
+                                if (max_entry_realtime >= rt_end) {
+                                        error(p,
+                                              "Entry realtime (%"PRIu64", %s) is too late with respect to tag (%"PRIu64", %s)",
```

Comment:
`Entry realtime` -> `The maximum entry realtime` is better?

Reply:
The fact that the maximum is too large implies that there exists an entry with a time that is too large. This entry is in conflict with the tag and that's what is reported here. I find this message more descriptive, instead of referring to a maximum which by itself does not exist in the journal file. We only report the discrepancy here instead of at the entry, because we only notice it, once we process the tag object. 

On the other hand I don't have a strong opinion on this and would adjust it to whatever you prefer.

Reply:
This is not strong opinion. Feel free to keep this as is.